src/services/users/update-user.js

```diff
@@ -0,0 +1,64 @@
+const db = require("../../db")
+const customError = require("../../lib/custom-error")
+const { UserNotFoundError } = require("../errors")
+
+/**
+ * 
+ * @param {Number} id - id of the user to be updated
+ * @param {Object} cols - columns to be updated
+ * @returns {String} - query to be executed
+ */
+const constructQuery = (cols) => {
+
+	// Setup static beginning of query
+	const query = ['UPDATE users']
+	query.push('SET')
+	// Create another array storing each set command
+	// and assigning a number value for parameterized query
+	const set = []
+	Object.keys(cols).forEach((key, i) => {
+	  set.push(`"${key}" = ($${(i + 1)})`) 
+	})
+	
+	query.push(set.join(', '))
+  
+	// Add the WHERE statement to look up by id
+	
+	query.push(`WHERE "id" = ($${set.length + 1}) RETURNING *` )
+  
+	// Return a complete query string
+	return query.join(' ')
+}
+
+/**
+ * 
+ * @param {Number} id - id of the user to be updated
+ * @param {Object} requestBody - request body
+ * @returns {Object} - updated user
+ */
+const updateUser = async (id,
+	requestBody
+) => {
+
+	const query = constructQuery(requestBody)	
+
+	const colValues = Object.keys(
+		requestBody
+	).map(key => requestBody[key])
```

Comment:
This is just me nitpicking and curious based on the choice used by the resource we used

I think we can use `Object.values` here in place of the `Object.keys` and `map` which is achieving the same thing

```suggestion
	const colValues = Object.values(
		requestBody
	)
```

or maybe that won't work for some reason?

Reply:
@jocrah changed